eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1622,6 +1622,17 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       d.commitments.resolveCommitment(tx) match {
         case Some(commitment) =>
           log.warning("a commit tx for fundingTxIndex={} fundingTxId={} has been confirmed", commitment.fundingTxIndex, commitment.fundingTxId)
+          // Funding transactions with a greater index will never confirm: we abandon them to unlock their wallet inputs,
+          // which would otherwise stay locked forever in our bitcoind wallet.
+          d.commitments.all
+            .collect { case c: Commitment if commitment.fundingTxIndex <= c.fundingTxIndex => c.fundingTxId }
+            .foreach { txId => wallet.abandon(txId) }
+          // Any anchor transaction that we created based on the latest local or remote commit will never confirm either
+          // so we need to abandon them to unlock their wallet inputs.
+          nodeParams.db.audit.listPublished(d.channelId).collect {
+            case tx if tx.desc == "local-anchor" => wallet.abandon(tx.txId)
```

Comment:
I think we want a _warning_ log whenever we call `abandon` as it's a pretty "risky" call.

Reply:
Good idea!

Reply:
Done in https://github.com/ACINQ/eclair/pull/2818/commits/0378fc4d785e550e7128f14044bdf8089344a17f